### PR TITLE
Merge to Master for v1.1.15

### PR DESCRIPTION
**Pull request by agause**
Release 1.1.15 makes changes necessary to support the Think Ministry MP oAuth updates that are scheduled to go into the INT environment on April 16, 2018. It updates the following things:

- Scopes now specified for  oAuth calls
- Update swagger integration to point to correct MP /authorize and /token endpoints.
 - The AuthUser class has an AdditionalProperties Dictionary that captures info that previously existed under the "additional_properties" object returned from the /oauth/me call; for ease of conversion, "ext_*" properties were mapped from the response onto this Dictionary, removing the "ext_" prefix before adding to the dictionary.

### CODE REVIEW

File: Crossroads.Web.Common/Constants.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿namespace Crossroads.Web.Common
+{
+    public static class Constants
+    {
+        /// <summary>
+        /// OAuth scopes url for Think Ministry.
+        /// </summary>
+        public const string ThinkMinistryOAuthScopesUrl = "http://www.thinkministry.com/dataplatform/scopes/all offline_access openid";
```
Comment (timmyg) [inline Crossroads.Web.Common/Constants.cs]: assuming the spaces in the url get encoded downstream somewhere?

Comment (timmyg) [reply Crossroads.Web.Common/Constants.cs]: also, is https endpoint available?

File: Crossroads.Web.Common/src/Security/AuthenticationRepository.cs [hunk 2]
```diff
@@ -217,6 +221,23 @@ public class AuthUser
         public List<string> Roles { get; set; }
         [JsonProperty("additional_properties")]
         public Dictionary<string, object> AdditionalProperties { get; set; }
+        [JsonExtensionData]
+#pragma warning disable 649
```
Comment (timmyg) [inline Crossroads.Web.Common/src/Security/AuthenticationRepository.cs]: these warning supposed to be here?

**Comment by timmyg** [comment]
I had a few small comments, logic and all looks good - looks like adding a client secret and calling a different endpoint for the most part? Did you guys test this out locally in a real project? I know theres a way you can point your nuget to a local version of a package

**Comment by timmyg** [comment]
Also wasnt sure if you guys were making additional changes or if the version should be bumped in this PR

**Comment by mjvcallibrity** [comment]
In the documentation we read, it explained that the PR from development into master was supposed to have a title that specified any version bumps. I suppose that is a good idea to also do when merging branch into development?

File: Crossroads.Web.Common/Constants.cs (hunk 1 above)
Comment (mjvcallibrity) [reply Crossroads.Web.Common/Constants.cs]: They must. All documentation from ThinkMinistry and notes by Jim indicate that it is space delimited, which seems really odd, but it the way it was presented to us. RE: the available endpoint, once again `http` and not `https` was presented to us in the documentation and notes, so we decided to use what was given to us.

File: Crossroads.Web.Common/src/Security/AuthenticationRepository.cs (hunk 2 above)
Comment (mjvcallibrity) [reply Crossroads.Web.Common/src/Security/AuthenticationRepository.cs]: Honestly...no clue. This was part of the work that Jim did before leaving. I don't know if it was something that he put in place as he tested things locally, or if needs to stay in.

**Comment by timmyg** [comment]
I was thinking there was an xml file somewhere in the project that specified the version, but couldnt find it anywhere when looking, so not really sure how you update the version just wanted to give a heads up that you'll have to do that somewhere

**Comment by mjvcallibrity** [comment]
Thanks! We'll be sure to figure that out before merging anything.

**Comment by timmyg** [comment]
Looks good.

Is the client secret required now? If so, should we remove the optionality of it as an argument in CreateRequest? https://goo.gl/aZQGC4

**Pull request merged**